cpp/src/arrow/acero/asof_join_node_test.cc

```diff
@@ -1770,5 +1771,38 @@ TEST(AsofJoinTest, DestroyNonStartedAsofJoinNode) {
       DeclarationToStatus(std::move(sink)));
 }
 
+// Reproduction of GH-46224: Hang when all left timestamps are greater than right
+// timestamps.
```

Comment:
For completeness, can we have a test for the reverse situation?

Reply:
Yeah, sure, will do.

Reply:
The hang doesn't occur for the reverse situation. But test updated anyway.